pkg/semtech-loramac/contrib/semtech_loramac.c

```diff
@@ -834,7 +834,14 @@ uint8_t semtech_loramac_send(semtech_loramac_t *mac, uint8_t *data, uint8_t len)
 
     _semtech_loramac_call(_send, &params);
 
-    return SEMTECH_LORAMAC_TX_SCHEDULE;
+    /* Wait for TX status information from the MAC */
```

Comment:
I think setting the `caller_pid` before calling `_semtech_loramac_call(...)` is missing, otherwise the msg never arrives
```C
mac->caller_pid = thread_getpid();
```

Reply:
Should be fixed now, can you try again @leandrolanzieri ? (I did and it works much better indeed :) )

Reply:
This fixes the issue. I tested using the example and the pkg test app, both on b-l072z-lrwan1 and lobaro-lorabox boards :+1: